<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Isoform detection using RNA seq de novo Assembly &mdash; WABI RNA-seq Workshop 0.1 documentation</title>
    
    <link rel="stylesheet" href="_static/sphinxdoc.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="WABI RNA-seq Workshop 0.1 documentation" href="index.html" />
    <link rel="next" title="Small RNA computer excercise" href="smallRNA-lab.html" />
    <link rel="prev" title="Identifying isoforms in RNA-seq data" href="isoform-lab.html" /> 
  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="smallRNA-lab.html" title="Small RNA computer excercise"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="isoform-lab.html" title="Identifying isoforms in RNA-seq data"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">WABI RNA-seq Workshop 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/scilife_logo.png" alt="Logo"/>
            </a></p>
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Isoform detection using RNA seq de novo Assembly</a><ul>
<li><a class="reference internal" href="#preparation">Preparation</a></li>
<li><a class="reference internal" href="#files-used-during-the-exercise">Files used during the exercise</a><ul>
<li><a class="reference internal" href="#if-you-are-on-uppmax">If you are on uppmax</a></li>
<li><a class="reference internal" href="#if-you-are-somewhere-else">If you are somewhere else</a></li>
</ul>
</li>
<li><a class="reference internal" href="#programs-used-during-the-exercise">Programs used during the exercise</a></li>
<li><a class="reference internal" href="#assemble-the-reads-into-contigs">Assemble the reads into contigs</a></li>
<li><a class="reference internal" href="#assessing-the-new-assemblies">Assessing the new assemblies</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="isoform-lab.html"
                        title="previous chapter">Identifying isoforms in RNA-seq data</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="smallRNA-lab.html"
                        title="next chapter">Small RNA computer excercise</a></p>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/isoform-denovo.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="isoform-detection-using-rna-seq-de-novo-assembly">
<h1>Isoform detection using RNA seq de novo Assembly<a class="headerlink" href="#isoform-detection-using-rna-seq-de-novo-assembly" title="Permalink to this headline">¶</a></h1>
<p>We are going to use one of the open source RNA <em>de novo</em> assemblers
during this practical. It is called <strong>Trinity</strong>. Independent assessment
of <em>de novo</em> assembly programs showed that Trinity was one of the best assemblers to use.
It is also one of the programs that is being updated and does also have downstream analysis tools.</p>
<img alt="_images/Compare.jpg" src="_images/Compare.jpg" />
<p>Figure taken from <a class="reference external" href="http://www.biomedcentral.com/1471-2105/12/S14/S2">Optimizing de novo transcriptome assembly from short-read RNA-Seq data: a comparative study</a>.</p>
<p>A de novo  take your reads and turn them into <em>contigs</em>. For more details
on how <strong>Trinity</strong> work read the corresponding paper (<a class="reference external" href="http://www.nature.com/nbt/journal/v29/n7/full/nbt.1883.html">Trinity</a>
).</p>
<p>Running an assembly using <strong>Trinity</strong> for ~20 million of reads takes at least a day and more than 50 GB of RAM. In order
to reduce the time for the <strong>Trinity</strong> to run during this course we will focus on reads that can be mapped to a small region on the human chromosome.</p>
<div class="section" id="preparation">
<h2>Preparation<a class="headerlink" href="#preparation" title="Permalink to this headline">¶</a></h2>
<p>If you are working on UPPMAX i suggest that you will do all your exercises from your glob folder.</p>
<p>Make a new subdirectory and go there for this exercise.</p>
<div class="highlight-python"><pre>#Go to your own glob folder mkdir deNovoAssembly
cd ~/glob/RNAseqWorkshop

#Create a new folder where you will do this exercise
mkdir deNovoAssembly
# go into that directory
cd deNovoAssembly</pre>
</div>
</div>
<div class="section" id="files-used-during-the-exercise">
<h2>Files used during the exercise<a class="headerlink" href="#files-used-during-the-exercise" title="Permalink to this headline">¶</a></h2>
<div class="section" id="if-you-are-on-uppmax">
<h3>If you are on uppmax<a class="headerlink" href="#if-you-are-on-uppmax" title="Permalink to this headline">¶</a></h3>
<p>Copy all the files that you will need for this exercise from here.</p>
<div class="highlight-python"><pre># This assumes that you now are in the deNovoAssembly folder
cp -r /proj/b2013006/webexport/downloads/courses/RNAseqWorkshop/deNovo/data .</pre>
</div>
</div>
<div class="section" id="if-you-are-somewhere-else">
<h3>If you are somewhere else<a class="headerlink" href="#if-you-are-somewhere-else" title="Permalink to this headline">¶</a></h3>
<p>You can download all data using a webinterface from <a class="reference external" href="https://export.uppmax.uu.se/b2013006/downloads/courses/RNAseqWorkshop/deNovo/">here</a></p>
</div>
</div>
<div class="section" id="programs-used-during-the-exercise">
<h2>Programs used during the exercise<a class="headerlink" href="#programs-used-during-the-exercise" title="Permalink to this headline">¶</a></h2>
<p>When doing this course on UPPMAX all programs will be available to load as pre-installed modules.
In order to be able to use these programs you need to load the modules before using them.</p>
<p>Load all programs that you will need for trinity to work on uppmax this exercise.</p>
<div class="highlight-python"><pre># load modules to make trinity work
module load bioinfo-tools
module load bowtie
module load samtools
module load trinity/2014-07-17</pre>
</div>
<p>Load all programs that you will need for STAR to work on UPPMAX this exercise.</p>
<div class="highlight-python"><pre># load modules to make RNAseq aligner STAR work

module load star</pre>
</div>
<p>If you are doing this exercise on somewhere else follow each program information on how to install it.</p>
</div>
<div class="section" id="assemble-the-reads-into-contigs">
<h2>Assemble the reads into contigs<a class="headerlink" href="#assemble-the-reads-into-contigs" title="Permalink to this headline">¶</a></h2>
<p>Since <strong>Trinity</strong> is often being updated you should make sure you are using the latest version.
That means that the requirements and the command line to to run <strong>Trinity</strong> changes occasionally.
You can find the basic usage info for <strong>Trinity</strong> <a href="#id1"><span class="problematic" id="id2">`</span></a>here
<a class="reference external" href="http://trinityrnaseq.sourceforge.net/#running_trinity">http://trinityrnaseq.sourceforge.net/#running_trinity</a></p>
<blockquote>
<div>and the latest typical command line to type <a href="#id3"><span class="problematic" id="id4">`</span></a>here</div></blockquote>
<p>&lt;<a class="reference external" href="http://trinityrnaseq.sourceforge.net/#running_trinity">http://trinityrnaseq.sourceforge.net/#running_trinity</a>&gt;`_</p>
<p>Go to the webpage and try out the code that is</p>
<p>Good things to know about the data used in this lab.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="c"># You will use paired end data.</span>
<span class="c"># The file that trinity refers to as left is the fastq files that ends with _1.fastq</span>
<span class="c"># The file that trinity refers to as right is the fastq files that ends with _2.fastq</span>

<span class="c"># The RNA seq data that we use in this exercise is not strand specific.</span>
</pre></div>
</div>
<p>To fully use the potential
of the programs it is worthwhile to read the manual and use the correct flags. As
an example both programs handle strand specific RNA that reduces the complexity of
the algorithm and also produces better results.
or <a class="reference external" href="http://trinityrnaseq.sourceforge.net/#running_trinity">Trinity</a>
manuals.</p>
</div>
<div class="section" id="assessing-the-new-assemblies">
<h2>Assessing the new assemblies<a class="headerlink" href="#assessing-the-new-assemblies" title="Permalink to this headline">¶</a></h2>
<p>Now that the reads have been assembled into contigs you can map them back onto
the human genome sequence to see how they were assembled. Note that in
non-model organism this is not possible. If you would like to asses the assembly
of transcripts without a reference genome Trinity has a downstream analysis pipeline
that is worth following <a class="reference external" href="http://trinityrnaseq.sourceforge.net/annotation/Trinotate.html">Trinotate</a> . This is not something we will
do in this course but if you have time over feel free to try it out.</p>
<p>Start with mapping the trinity assembled transcripts to the human genome using STAR.
Convert them to bam format, sort and index them using samtools:</p>
<div class="highlight-python"><pre>mkdir STARtrinityMapping

STAR  --genomeDir /proj/g2014046/private/RNAseqWorkshop/reference/hg19_Gencode14.overhang75  --readFilesIn Trinity/Trinity.fasta --runThreadN 1 --outSAMstrandField intronMotif --outFileNamePrefix STARtrinityMapping/
samtools view -bSh -o trinityTranscripts.bam STARtrinityMapping/Aligned.out.sam
samtools sort trinityTranscripts.bam  trinityTranscripts.sorted
samtools index trinityTranscripts.sorted.bam</pre>
</div>
<p>Do the same procedure for the oases assembled transcripts:</p>
<div class="highlight-python"><pre>mkdir STARoasesMapping
STAR  --genomeDir /proj/g2014046/private/RNAseqWorkshop/reference/hg19_Gencode14.overhang75  --readFilesIn oasesPipelineMerged/transcripts.fa --runThreadN 1 --outSAMstrandField intronMotif --outFileNamePrefix STARoasesMapping/
samtools view -bSh -o oasesTranscripts.bam STARoasesMapping/Aligned.out.sam
samtools sort oasesTranscripts.bam  oasesTranscripts.sorted
samtools index oasesTranscripts.sorted.bam</pre>
</div>
<p>When ready there should be two bam files that are sorted and indexed. These can now be viewed in the IGV
or Savant genome browsers. In total there were 12 samples and you have now assembled one of those samples.
If time permits do one more sample. If time is running out you can download and view all the 24 different samples.
We have also merged the reads from all the 12 samples and used all the reads to create assembled transcripts.
All these files can be found <a class="reference external" href="https://export.uppmax.uu.se/g2014046/files/RNAseqWorkshop/download/RNAseq/deNovoFinishedFiles/AllBamFiles/">here</a></p>
<p>Download a few of them and compare the differents states to see if you can identify different isoforms. How does the
de novo assembled transcripts compare to the reference based isoform detection programs.</p>
<p><strong>OPTIONAL</strong>
I recomend to download the bamFiles and view them in a genome browser on your laptop.
The interactive genome view experience on UPPMAX, especially when loading many tracks, can
be slow.This is done in two steps.</p>
<div class="highlight-python"><pre>#create a folder for all the bamfiles
mkdir AllBamFiles

# move all the bamfiles into that folder
mv *.sorted.bam AllBamFiles

#create a tar file with all the bamFiles so that you can download them to your laptop

tar -cf AllBamFiles.tar AllBamFiles

#Use any sftp program of your choice to download the files from uppmax

# If you are using shell you can open up a new terminal window and go to
# the place where you want to store your bamFiles

cd $YOURLOCALPATH
scp yourUppmaxName@milou.uppmax.uu.se:$WORKDIR/deNovo/AllBamFiles.tar .</pre>
</div>
<p>Now that you have all the bam files in with individual names try to view them in
a genome brower, both IGV and Savant works fine. Here we will describe how to view them
in IGV but SAVANT has a nice feature of viewing paired end reads as arcs that IGV
misses. If you have time i recomend trying both of them out.</p>
<p>First have a look on the  two bamfiles that contains the assemblies of all
reads from all twelve timepoints with the two different assemblers. They have the
names <tt class="docutils literal"><span class="pre">RAB11FIP5_trinity.Trinity._hg_19_STAR.bam</span></tt> and
<tt class="docutils literal"><span class="pre">RAB11FIP5_oases.Oases._hg_19_STAR.bam</span></tt>.</p>
<div class="highlight-python"><pre>#If you view your files on your laptop start IGV like this

java -Xmx1500M -jar igv.jar

# If you view your files on UPPMAX do according to UPPMAX


#Load tracks in the IGV browser

File-&gt;Load From File...
    choose **oasesTranscripts.sorted.bam**
    and **trinityTranscripts.sorted.bam**

# Load peptide sequences

File-&gt;Load From File...
    choose **human_A431_global-TDA-FDR1pc_green-known_red-novel.bed**

# Load your mapped reads from before

File-&gt;Load From File...
    choose **sample12_RAB11FIP5.bam**

# Load your own GTF file

File-&gt;Load From File...
    choose **transcripts.gtf** or what you have named it.</pre>
</div>
<p><strong>OPTIONAL</strong>
There is also a possibility to view tracks that is publicly available. This is easy to
do in IGV and adds some information in the region that we are looking into.</p>
<div class="highlight-python"><pre> # Load different gene annotations files

 File-&gt;Load From Server...
     choose Available Datasets -&gt;Annotations -&gt; Genes -&gt;UCSC Genes


 # Load multiple alignments to other vertebrates

 File-&gt;Load From Server...
     choose Available Datasets -&gt;Annotations -&gt; Comparative Genomics -&gt;Phastcons (Vertebrate 46 way)


# Load any of the other annotations that you think is interesting

 File-&gt;Load From Server...
     choose Available Datasets -&gt;..  -&gt; ..  -&gt;Up to you</pre>
</div>
<p>Now have a look at the de novo assembled transcripts. Do they seem reasonable? Which
regions on the de novo assembled transcripts do not correspond to your own .gtf
file?  Which is the correct one?</p>
<p>Now take a closer look at the region chr2:73,308,166-73,308,278. This corresponds
to the regions where the RefSeq genes is annotated as intron but the <em>de novo</em> assembly
, the cufflinks gtf file and the peptide file suggest that the region is being transcribed
and translated into peptides. When examining the <em>de novo</em> assembled contigs it seems
that none of the transcripts goes through the region. Is this real or could there
be a shortcoming of the assembler or the sequencing platform? Unfortunately we do
not have the answers to these questions but all the different methods add in to give
more understanding in the complexity of isoform analysis and genome annotation.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="smallRNA-lab.html" title="Small RNA computer excercise"
             >next</a> |</li>
        <li class="right" >
          <a href="isoform-lab.html" title="Identifying isoforms in RNA-seq data"
             >previous</a> |</li>
        <li><a href="index.html">WABI RNA-seq Workshop 0.1 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer">
        &copy; Copyright 2013, Johan Reimegård, Mikael Huss, and Sanela Kjellqvist.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2b1.
    </div>
  </body>
</html>